### ISSUE TITLE
Merge plone.theme into Products.CMFPlone

### ISSUE THREAD

**Issue by jensens** [plone/Products.CMFPlone#1934]
- one package with 43 LOC - phew...
- skin layers should be deprecated anyway, this can go at some point.

**Comment by gforcada** [comment]
Then we should at least mark them for deprecation already, so everyone is already aware of this decision :-)

**Comment by ArhantJain** [comment]
@jensens @gforcada  I want to work on this issue. Please guide me the way to sort out this.

**Comment by ArhantJain** [comment]
Hii @jensens 
I understand the issue you are depicting. 
So, Please guide me the way of solving this and where should I start from?

**Comment by jensens** [comment]
In fact its about moving the code from plone.theme to a place in this package and properly deprecate plone.theme. From my smartphone its hard to tell details, I am back in office Monday to follow up on this. Thanks for taking this task!

**Comment by ArhantJain** [comment]
Okay, I will wait for your guidelines. :)

**Comment by jensens** [comment]
@ArhantJain thanks for taking this task.

This is work for Plone 5.2, since 5.1 ist released in and in maintenance mode .

This package just provides one Interface in `interfaces.py`and a subscriber in ``layer.py``. 

Move the code (to be done on `Products.CMFPlone` master based feature branch  first):
- the ``interfaces.py`` needs to be moved to a new ``Products/CMFPlone/interfaces/defaultlayer.py``. 
- the ``layer.py`` needs to be moved to ``Products/CMFPlone/marklayer.py``. 

Current ``plone.theme`` need to be branched away to a 3.x branch. On master the version number increases to 4.0 (`setup.py`/`CHANGES.rst`). Then on master:
- the ``layer.py`` can be deleted. 
- the ``interfaces.py`` needs a deprecation fallback (backward compatibility, BBB). See [Plone Deprecation Guide](https://docs.plone.org/develop/styleguide/deprecation.html#moving-functions-and-classes) for details.
- in `buildout.coredev` branch 5.1 in the `sources.cfg`` the branch of `plone.theme` need to be set to 3.x (for buildout.coredev 5.2 we stay on master)

Now the tests needs to be moved to `Products.CMFPlone.tests`.
- testing.py is not needed any longer, we can use the layers from Products.CMFPlone.
- all `plone/theme/tests/*` goes - renamed to meaningful names - to `Products/CMFPlone/tests`. 
- now tests need to be modified to test if the interface can be imported form the old and new place (BBB).

All work to be done on a branch, one for `plone.theme` and one for `Products.CMFPlone`.

In fact this is an easy task because you don't need to modify much code. Just the fact you have to juggle with two branches and BBB makes it a bit complex. Latter needs first some oversight how to work with the buildout.coredev. I would first try get buildout.coredev up and running for 5.1 and 5.2, then create features branches in the buildout.coredev 5.2 version for Products.CMFPlone and plone.theme abd start working. 

If you have any questions come back here.

@gforcada Did I miss something?

**Comment by gforcada** [comment]
@jensens yes you missed the point that this is by no means easy... even though code can be moved verbatim from one place to the other, ensuring that this movement is done without breaking everything apart, deprecation notices and so on, are far too much for a newcomer IMHO. But hey, if @ArhantJain is brave enough to go through it I will shut up right now :no_mouth: 

Back to the point, create branches on top of master for both packages and start implementing the changes outlined by @jensens . Let us know where are you stuck and try to do some searches and see how other packages are outlined (@jensens any pointers for other packages being deprecated that he can use as a reference?)

**Comment by ArhantJain** [comment]
Thanks, @jensens  @gforcada... I will follow this.

Sorry, for the late reply. Semester Exams of my University are going on.
I will be on the issue just after two days.

Thanks :)

**Comment by jensens** [comment]
Closing as outdated (9 years old). plone.theme is still maintained as a separate package (5.0.0a1). After 9 years the merge is unlikely to happen and the cost/benefit does not justify the BBB complexity.
If you think this issue is still relevant please reopen it. Thank you for the original report!
